integration/container/resize_test.go

```diff
@@ -0,0 +1,100 @@
+package container
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	req "github.com/docker/docker/integration-cli/request"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResizeAPIResponse(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	c, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{},
+		&network.NetworkingConfig{},
+		"",
+	)
+	require.NoError(t, err)
+
+	err = client.ContainerStart(ctx, c.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+
+	poll.WaitOn(t, containerIsInState(ctx, client, c.ID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	err = client.ContainerResize(ctx, c.ID, types.ResizeOptions{
+		Height: 40,
+		Width:  40,
+	})
+	require.NoError(t, err)
+}
+
+func TestResizeAPIHeightWidthNoInt(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	c, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{},
+		&network.NetworkingConfig{},
+		"",
+	)
+	require.NoError(t, err)
+
+	err = client.ContainerStart(ctx, c.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+
+	poll.WaitOn(t, containerIsInState(ctx, client, c.ID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	endpoint := "/containers/" + c.ID + "/resize?h=foo&w=bar"
+	res, _, err := req.Post(endpoint)
+	require.NoError(t, err)
+	assert.Equal(t, res.StatusCode, http.StatusBadRequest)
+}
+
+func TestResizeAPIResponseWhenContainerNotStarted(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	c, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"true"},
+			Image: "busybox",
+		},
+		&container.HostConfig{},
+		&network.NetworkingConfig{},
+		"",
+	)
+	require.NoError(t, err)
+
+	err = client.ContainerStart(ctx, c.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+
+	poll.WaitOn(t, containerIsInState(ctx, client, c.ID, "exited"), poll.WithDelay(100*time.Millisecond))
+
+	err = client.ContainerResize(ctx, c.ID, types.ResizeOptions{
+		Height: 40,
+		Width:  40,
+	})
+	assert.Contains(t, err.Error(), "is not running")
```

Comment:
This will panic if `err` is nil, so you'll  need to check that first. We have a `testutil.ErrorContains()` for this I think.

Reply:
Done. Thanks.